src/common/darktable_ucs_22_helpers.h

```diff
@@ -233,14 +233,15 @@ static inline struct dt_iop_order_iccprofile_info_t * D65_adapt_iccprofile(struc
   if(work_profile) // && !isnan(work_profile->matrix_in[0][0]))
   {
     // Alloc
-    struct dt_iop_order_iccprofile_info_t *white_adapted_profile = (dt_iop_order_iccprofile_info_t *)malloc(sizeof(dt_iop_order_iccprofile_info_t));
+    struct dt_iop_order_iccprofile_info_t *white_adapted_profile =
+      (dt_iop_order_iccprofile_info_t *)dt_alloc_aligned(sizeof(dt_iop_order_iccprofile_info_t));
 
     // Init a new temp profile by copying the base profile
     memcpy(white_adapted_profile, work_profile, sizeof(dt_iop_order_iccprofile_info_t));
 
     // Multiply the in/out matrices by the chromatic adaptation matrix
-    dt_colormatrix_t input_matrix;
-    dt_colormatrix_t output_matrix;
+    dt_colormatrix_t DT_ALIGNED_ARRAY input_matrix;
+    dt_colormatrix_t DT_ALIGNED_ARRAY output_matrix;
```

Comment:
The definition of `dt_colormatrix_t` already includes the DT_ALIGNED_ARRAY, so these don't actually do anything.  But the aligned allocation is needed for any embedded fields in `dt_iop_order_iccprofile_info_t` which need to be aligned.

Reply:
Confirmed I cannot reproduce the crash with just the aligned allocation. Thanks!